resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/HttpClientUtil.scala

```diff
@@ -42,6 +48,49 @@ private[spark] object HttpClientUtil {
     Option.apply(trustContext).foreach(context => {
       httpClientBuilder = httpClientBuilder.sslSocketFactory(sslSocketFactory, context)
     })
+    val uriObjects = uris.map(URI.create)
+    val httpUris = uriObjects.filter(uri => uri.getScheme == "http")
+    val httpsUris = uriObjects.filter(uri => uri.getScheme == "https")
+    val maybeAllProxy = Option.apply(System.getProperty(Config.KUBERNETES_ALL_PROXY))
```

Comment:
We should keep it at this level to ensure that we're always picking up the properties whenever we set up HTTP clients. In other words, this class is opinionated towards using the system properties to set up the proxy. If we wanted to allow for multiple opinions, for example also allowing the proxy configuration to be set via `SparkConf`, then we should inject these values from the caller.